internal/pkg/cli/storage_init.go

```diff
@@ -0,0 +1,237 @@
+}
+
+type initStorageVars struct {
+	*GlobalOpts
+	StorageType string
+	StorageName string
+	StorageSvc  string
+}
+
+type initStorageOpts struct {
+	initStorageVars
+
+	fs    afero.Fs
+	ws    wsSvcReader
+	store store
+
+	app *config.Application
+}
+
+func newStorageInitOpts(vars initStorageVars) (*initStorageOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store client: %w", err)
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace client: %w", err)
+	}
+
+	return &initStorageOpts{
+		initStorageVars: vars,
+
+		fs:    &afero.Afero{Fs: afero.NewOsFs()},
+		store: store,
+		ws:    ws,
+	}, nil
+}
+
+func (o *initStorageOpts) Validate() error {
+	if o.AppName() == "" {
+		return errNoAppInWorkspace
+	}
+	if o.StorageSvc != "" {
+		if err := o.validateServiceName(); err != nil {
+			return err
+		}
+	}
+
+	if o.StorageType != "" {
+		if err := validateStorageType(o.StorageType); err != nil {
+			return err
+		}
+	}
+
+	if o.StorageName != "" {
+		var err error
+		switch o.StorageType {
+		case dynamoDBStorageType:
+			err = dynamoTableNameValidation(o.StorageName)
+		case s3StorageType:
+			err = s3BucketNameValidation(o.StorageName)
+		default:
+			// use dynamo since it's a superset of s3
+			err = dynamoTableNameValidation(o.StorageName)
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *initStorageOpts) Ask() error {
+	if err := o.askStorageSvc(); err != nil {
+		return err
+	}
+	if err := o.askStorageType(); err != nil {
+		return err
+	}
+	if err := o.askStorageName(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *initStorageOpts) askStorageType() error {
+	if o.StorageType != "" {
+		return nil
+	}
+
+	storageType, err := o.prompt.SelectOne(fmt.Sprintf(
+		fmtStorageInitTypePrompt, color.HighlightUserInput(o.StorageSvc)),
+		storageInitTypeHelp,
+		storageTypes)
+	if err != nil {
+		return fmt.Errorf("select storage type: %w", err)
+	}
+
+	o.StorageType = storageType
+	return nil
+}
+
+func (o *initStorageOpts) askStorageName() error {
+	if o.StorageName != "" {
+		return nil
+	}
+	var validator func(interface{}) error
+	switch o.StorageType {
+	case dynamoDBStorageType:
+		validator = dynamoTableNameValidation
+	case s3StorageType:
+		validator = s3BucketNameValidation
+	}
+	name, err := o.prompt.Get(fmt.Sprintf(fmtStorageInitNamePrompt,
+		color.HighlightUserInput(o.StorageType)),
+		storageInitNameHelp,
+		validator)
+
+	if err != nil {
+		return fmt.Errorf("input storage name: %w", err)
+	}
+	o.StorageName = name
+	return nil
+}
+
+func (o *initStorageOpts) askStorageSvc() error {
+	if o.StorageSvc != "" {
+		return nil
+	}
+	localSvcNames, err := o.ws.ServiceNames()
+	if err != nil {
+		return fmt.Errorf("retrieve local service names: %w", err)
+	}
+	svc, err := o.prompt.SelectOne(storageInitSvcPrompt,
+		storageInitSvcHelp,
+		localSvcNames,
+	)
+	o.StorageSvc = svc
+	return nil
+}
```

Comment:
Can you use the https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/cli/selector/selector.go#L71 service selector?